modular_skyrat/master_files/code/modules/pain_shock/status_indicator.dm

```diff
@@ -0,0 +1,198 @@
+		return TRUE
+
+/mob/living/proc/is_unconcious()
+	if(HAS_TRAIT(src, TRAIT_KNOCKEDOUT))
+		return TRUE
+
+/mob/living/proc/is_confused()
+	if(has_status_effect(/datum/status_effect/confusion))
+		return TRUE
+
+/* /mob/living/carbon/proc/is_blind_status()
+	if(HAS_TRAIT_FROM(src, TRAIT_BLIND, STATUS_EFFECT_TRAIT) || HAS_TRAIT_FROM(src, TRAIT_BLIND, EYES_COVERED) && !HAS_TRAIT_FROM(src, TRAIT_BLIND, QUIRK_TRAIT)) // We don't want permanently blind users (those who took the trait) to have this forever.
+		return TRUE */ // Disabled, uncomment if we want this.
+
+/mob/living/carbon/death(gibbed) // On death, we clear the indiciators
+	..() // Call the TG death. Do not . = ..()!
+	for(var/iteration in status_indicators) // When we die, clear the indicators.
+		remove_status_indicator(icon_state) // The indicators are named after their icon_state and type
+/* /datum/status_effect/incapacitating/on_apply()
+	. = ..()
+	addtimer(CALLBACK(owner, /mob/living/.proc/status_sanity), 100 MILLISECONDS)
+/datum/status_effect/incapacitating/on_remove()
+	. = ..()
+	addtimer(CALLBACK(owner, /mob/living/.proc/status_sanity), 1000 MILLISECONDS) */
+
+/* /mob/living/carbon/apply_effect()
+	. = ..()
+	status_sanity() */
+
+/mob/living/carbon/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_CARBON_HEALTH_UPDATE, .proc/status_sanity)
+
+/* /mob/living/updatehealth()
+	. = ..()
+	status_sanity() */
+/mob/living/proc/status_sanity()
+	SIGNAL_HANDLER
+	if(stat == DEAD)
+		for(var/iteration in status_indicators) // When we die, clear the indicators.
+			remove_status_indicator(icon_state) // The indicators are named after their icon_state and type
+	is_weakened() ? add_status_indicator("weakened") : remove_status_indicator("weakened")
+	is_paralyzed() ? add_status_indicator("paralysis") : remove_status_indicator("paralysis")
+	is_stunned() ? add_status_indicator("stunned") : remove_status_indicator("stunned")
+	is_unconcious() ? add_status_indicator("sleeping") : remove_status_indicator("sleeping")
+	is_confused() ? add_status_indicator("confused") : remove_status_indicator("confused")
+//	is_blind_status() ? add_status_indicator("blinded") : remove_status_indicator("blinded") // Disabled, keeping for IF we decide we want this.
+//	is_critical() ? add_status_indicator("critical") : remove_status_indicator("critical") // Also disabled. Crosses medhud territory, despite being a shift click examine thing.
+
+/mob/living/proc/add_status_indicator(image/thing)
+	if(get_status_indicator(thing)) // No duplicates, please.
+		return
+	if(!istype(thing, /image))
+		thing = image(icon = 'modular_skyrat/master_files/icons/mob/status_indicators.dmi', loc = src, icon_state = thing)
```

Comment:
From what I could see so far, the parameter `thing` (which should ***definitely*** have a better name, like `indicator_name`) is never an image, and always seems to be a string. Thus, this if() statement feels pointless, because it's always going to be the icon_state that you pass to it.

I feel like those images could be cached, and then added as overlays on the mob afterwards, so basically you'd likely go from having up to like five images for every mob, down to five images in total, if it truly works like I'm feeling it does. I'll ask around to see if that's how it works, but you can still try that in the meantime, if that's faster for you.

Reply:
I asked around on /tg/, and apparently, yes, it's possible to add the same image as an overlay to many different mobs. So this is what I'd like you to do, that way there's not an ever-increasing memory cost tied to these indicators, based on the amount of mobs that exist, but rather a flat cost of up to 5 images for the round. I think we can both agree that's a very good thing for performance.

Reply:
renamed next commit, not sure how to do that